vars/publishToNpm.groovy

```diff
@@ -7,16 +7,30 @@
  * compatible open source license.
  */
 
-/** Library to publish artifacts to NPM registry under @opensearch-project namespace
+/** Library to publish artifacts to NPM registry under @opensearch-project namespace.
 @param Map args = [:] args A map of the following parameters
-@param args.repository <required> - Repository to be used to publish the artifact to npm
-@param args.tag <required> - Tag reference to be used to publish the artifact
+@param args.repository <required> - Repository to be used to publish the artifact to npm.
+@param args.tag <required> - Tag reference to be used to publish the artifact.
+OR
+@param args.artifactPath <required> - Artifact path to publish to NPM. Defaults to package.json See supported artifacts https://docs.npmjs.com/cli/v9/commands/npm-publish?v=true#description for more details.
 */
 void call(Map args = [:]) {
+    artifactPath = args.artifactPath ?: ''
 
-    checkout([$class: 'GitSCM', branches: [[name: "${args.tag}" ]], userRemoteConfigs: [[url: "${args.repository}" ]]])
-    
-    withCredentials([string(credentialsId: 'jenkins-opensearch-publish-to-npm-token', variable: 'NPM_TOKEN')]){
-        sh """npm set registry "https://registry.npmjs.org"; npm set //registry.npmjs.org/:_authToken ${NPM_TOKEN}; npm publish --dry-run && npm publish --access public"""
+    if (args.artifactPath && args.repository && args.tag) {
+        currentBuild.result = 'ABORTED'
+        error('Artifact Path and GitHub repository details are mutually exclusive. Please use either repository and tag OR artifactPath as arguments')
     }
-}
\ No newline at end of file
+    if (!args.artifactPath) {
+        checkout([$class: 'GitSCM', branches: [[name: "${args.tag}" ]], userRemoteConfigs: [[url: "${args.repository}" ]]])
+    }
+
+    withCredentials([string(credentialsId: 'jenkins-opensearch-publish-to-npm-token', variable: 'NPM_TOKEN')]) {
+        sh """
```

Comment:
Might want to use nvm to switch node/npm version just in case a bump to the image, then cause issues during the publish.

Reply:
Node is not in picture at all since we are not building anything on jenkins so we are good for now.

Reply:
I am actually not sure if nvm use change to a different node, does npm change as well or not.

Reply:
I dont think so. since publishing is not as such building anything any node should be fine. Has not caused any issues yet but will keep watch.